docs/blog/2019-04-18-component-shadowing/index.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Component Shadowing
+date: 2019-04-18
+author: John Otander
+excerpt: "Gatsby Themes introduce a concept called Component Shadowing. This feature allows
+users to override a component in order to customize its rendering."
+tags:
+  - themes
+---
+
+Gatsby Themes introduce a concept called Component Shadowing. This feature allows
+users to override a component in order to customize its rendering. Component Shadowing
+is a powerful way to make one off changes to a theme without having to fully eject.
+
+It's pretty common to end up with a theme where you're happy with 99% of the details
+but want to customize the color palette and perhaps the author bio. This is where Component
+Shadowing is an effective way to make changes quickly and intuitively.
+
+## Shadowing Example
+
+If you've installed `gatsby-theme-blog` you'll notice that it renders a
+[Bio component](https://github.com/gatsbyjs/gatsby/blob/666a9bc3c8d91be8a3118b1128340a06e895735e/themes/gatsby-theme-blog/src/components/bio.js)
+which is used in the BlogPost template. If you'd like to change the component you can do so
+with Component Shadowing.
+
+### Theme File Structure
+
+`gatsby-theme-blog` has the following file structure (note that some files are
+omitted for brevity):
+
+```
+gatsby-theme-blog
+├── package.json
+├── src
+│   ├── components
+│   │   ├── bio.js
+│   │   ├── layout.js
+│   │   ├── profile-pic.jpg
+│   │   └── tokens
+│   │       └── index.js
+│   ├── gatsby-theme-blog-core
+│   │   └── components
+│   │       └── blog-post.js
+│   └── utils
+│       └── typography.js
+```
+
+### Implementing a Shadow
+
+Component Shadowing uses a naming convention to determine which component will be rendered.
+So, in order to override the Bio component in `gatsby-theme-blog` you'd create a file named
+`src/gatsby-theme-blog/components/bio.js`.
+
+Any file that lives in `src/gatsby-theme-blog/` of the user's site will be used _instead_ of a
+file with the same name in `gatsby-theme-blog/src`.
```

Comment:
This is a little confusing. Should the second mention be `gatsby-theme-blog-core`?

Reply:
The file paths here are correct and it could be useful to include `<user-site>` or something on the beginning of the first path. The first path is a path to a directory in the user's site and the second is the theme npm package.

maybe this is better?

> Any file that lives in `src/gatsby-theme-blog/` of the user's site will be used _instead_ of a file with the same name in the theme's src directory: `gatsby-theme-blog/src`.

Reply:
I think I was thrown off by the file diagram above this section, which references `gatsby-theme-blog-core`. Is that accurate with this description?

Reply:
ah I see why you were thrown off. That is definitely confusing. It's additional information that isn't necessary to understand the `gatsby-theme-blog` description here.

In addition to what we are talking about (which is the user's site shadowing components from `gatsby-theme-blog`), `gatsby-theme-blog` shadows components from `gatsby-theme-blog-core`. This is why there is a file located at `gatsby-theme-blog/src/gatsby-theme-blog-core/components/blog-post.js`.

I don't see any other mentions of the `-core` theme and we aren't really talking about Latent Shadowing here, so perhaps we should remove that section of the `tree` output @johno as it doesn't seem relevant?